### PR TITLE
Filter out disabled REST methods from the OpenAPI document

### LINKED ISSUE #34030: REST endpoints disabled via the `@EndpointDisabled` annotation still appear in swagger
### Describe the bug

The `@EndpointDisabled` annotation conditionally disables REST endpoints, but it is controlled by a runtime property, so we can't filter out affected endpoints at build time. We will need a runtime filter that dynamically hides these endpoints from swagger-ui

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jmartisk**
Fixes #34030

### CODE REVIEW

File: extensions/smallrye-openapi/deployment/pom.xml [hunk 1]
```diff
@@ -69,7 +69,7 @@
     <!-- Test -->
     <dependency>
       <groupId>io.quarkus</groupId>
-      <artifactId>quarkus-resteasy-deployment</artifactId>
+      <artifactId>quarkus-resteasy-reactive-deployment</artifactId>
```
Comment (geoand) [inline extensions/smallrye-openapi/deployment/pom.xml:72]: Is smallrye-openapi supposed to have such a hard dependency?

Comment (jmartisk) [reply extensions/smallrye-openapi/deployment/pom.xml:72]: There's no hard dependency, it's `<scope>test</scope>`. Here I'm merely migrating the test suite to use resteasy-reactive instead of classic.

Comment (geoand) [reply extensions/smallrye-openapi/deployment/pom.xml:72]: Ah okay, it was not clear from the Github UI

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java [hunk 2]
```diff
@@ -127,6 +139,27 @@ public BeanFactory.BeanInstance<?> apply(Class<?> aClass) {
             }
             Map<String, RequestMapper<RuntimeResource>> mappersByMethod = new RuntimeMappingDeployment(templates)
                     .buildClassMapper();
+            for (Map.Entry<String, RequestMapper<RuntimeResource>> entry : mappersByMethod.entrySet()) {
```
Comment (geoand) [inline independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java:142]: I'd like to avoid iterating over entries of a map. There are some examples in this class that use Map.forEach instead

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java [hunk 3]
```diff
@@ -104,17 +104,29 @@ public BeanFactory.BeanInstance<?> apply(Class<?> aClass) {
...
+        // end with one
+        String prefix = rootPath;
+        if (prefix != null) {
+            prefix = sanitizePathPrefix(prefix);
+        } else {
+            prefix = "";
+        }
+        if ((applicationPath != null) && !applicationPath.isEmpty()) {
+            prefix = prefix + sanitizePathPrefix(applicationPath);
+        }
```
Comment (geoand) [inline independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java:118]: Why is this needed?

Comment (jmartisk) [reply independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java:118]: I've just moved this piece of code a bit further up, because I need the prefix to compute the full resource paths, including the `quarkus.http.root-path` if it's defined.

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java (hunk 2 above)
Comment (jmartisk) [reply independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java:142]: I remember you being against using lambda expressions in runtime code (at some point in the past anyway), so I think I avoided forEach basically for that reason ;)

Comment (geoand) [reply independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java:142]: In general yes, but the fact is that iterating of maps is even worse :(. `Map.forEach` allows for optimal map access and when used with method references should lead to minimal metaspace overhead

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java (hunk 3 above)
Comment (geoand) [reply independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java:118]: Ah, thanks!

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java (hunk 2 above)
Comment (jmartisk) [reply independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java]: Ok, changed that to `forEach`

**Review by geoand — APPROVED** [review]

**Comment by manofthepeace** [comment]
Could this be backported to 3.2?

**Comment by jmartisk** [comment]
I do think so. Added the backport label.
